extensions/amp-base-carousel/1.0/arrow.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import * as styles from './base-carousel.css';
+import {dict} from '../../../src/utils/object';
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function Arrow(props) {
+  const {
+    'customArrow': customArrow,
+    'dir': dir,
+    'onClick': onClick,
+    'disabled': disabled,
+  } = props;
+  const button = customArrow
+    ? customArrow
+    : renderDefaultArrow(dict({'dir': dir}));
+  const {
+    'children': children,
+    'disabled': customDisabled,
+    'onCustomClick': onCustomClick,
+    ...rest
+  } = button.props;
+  const isDisabled = disabled || customDisabled;
+  const handleClick = (e) => {
+    if (onCustomClick) {
+      onCustomClick(e);
+    }
+    onClick();
+  };
+  return (
+    <div
+      style={{
+        ...styles.arrowPlacement,
+        // Offset button from the edge.
+        [dir < 0 ? 'left' : 'right']: '0px',
+      }}
+    >
+      {Preact.cloneElement(
+        button,
+        {
+          onClick: handleClick,
```

Comment:
Should this also be `'onClick'`?

Reply:
Can you clarify the question -- do you mean to add single quotes around the property, or to rename the value `handleClick` to `onClick`? If the latter, `handleClick` calls both `onClick` (passed from `BaseCarousel` to `setState`) and `onCustomClick` (passed from user-supplied custom arrow) -- I can rename these to reduce confusion.

Reply:
Yes. I'm wondering if all props passed to `Preact.cloneElement` or `Preact.createElement` should just always have quotes for keys? /cc @jridgewell

Reply:
Done.